tracee-ebpf/tracee/process_tree.go

```diff
@@ -0,0 +1,225 @@
+package tracee
+
+import (
+	"bufio"
+	"encoding/binary"
+	"fmt"
+	"github.com/aquasecurity/tracee/pkg/containers"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	_ "github.com/aquasecurity/tracee/pkg/containers"
+	"github.com/aquasecurity/tracee/pkg/external"
+)
+
+type ProcessTree struct {
+	processTreeMap map[int]external.ProcessCtx
+}
+
+func SetProcessContext(ctx []byte) external.ProcessCtx {
+	var procCtx = external.ProcessCtx{}
+	procCtx.StartTime = int(binary.LittleEndian.Uint64(ctx[0:8]))
+	procCtx.ContainerID = fmt.Sprintf("%v", binary.LittleEndian.Uint64(ctx[8:16]))
+	procCtx.Pid = int(binary.LittleEndian.Uint32(ctx[16:20]))
+	procCtx.Tid = int(binary.LittleEndian.Uint32(ctx[20:24]))
+	procCtx.Ppid = int(binary.LittleEndian.Uint32(ctx[24:28]))
+	procCtx.HostPid = int(binary.LittleEndian.Uint32(ctx[28:32]))
+	procCtx.HostTid = int(binary.LittleEndian.Uint32(ctx[32:36]))
+	procCtx.HostPpid = int(binary.LittleEndian.Uint32(ctx[36:40]))
+	procCtx.Uid = int(binary.LittleEndian.Uint32(ctx[40:44]))
+	procCtx.MntId = int(binary.LittleEndian.Uint32(ctx[44:48]))
+	procCtx.PidId = int(binary.LittleEndian.Uint32(ctx[48:52]))
+	return procCtx
+}
+
+func timespecToTime(ts syscall.Timespec) time.Time {
+	return time.Unix(int64(ts.Sec), int64(ts.Nsec))
+}
+
+func getFileStartTime(path string) (time.Time, error) {
+	var ctime time.Time
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		return ctime, err
+	}
+	stat_t := fileInfo.Sys().(*syscall.Stat_t)
+	return timespecToTime(stat_t.Ctim), nil
+}
+
+func parseProcStatus(status []string, taskName string) (external.ProcessCtx, error) {
+	var process external.ProcessCtx
+	processFields := []string{"Tgid", "Pid", "PPid", "Uid", "NStgid", "NSpid", "NSpgid"}
+	processVals := make(map[string]int)
+	i := 0
+	//var processVals []int
+	for idx, val := range status {
+		if val == processFields[i] {
+			i++
+			field, err := strconv.ParseUint(status[idx+1], 10, 32)
+			if err != nil {
+				return process, err
+			}
+			processVals[processFields[i]] = int(field)
+		}
+		if i > len(processFields)-1 {
+			break
+		}
+	}
+	processCtime, err := getFileStartTime(taskName)
```

Comment:
processCtime -> processStartTime
Also, is this the real process start time as represnted by the kernel?
You can get the real process start time from /proc/pid/stat

Reply:
yes and no, i looks in /proc/pid/stat and /proc/pid/status for the process Ctime but i couldn't find it. 
so i can up with idea to take the directory Ctime instead due the reason its created in the task creation.
if u willing to get the exact Ctime i can give another look on that but when i tested it it seemed to work fine

Reply:
Btw afaik the kernel itself create this dirs in the process creation so i don't think there is to supposed to be a margin between the times

Reply:
Look for starttime in https://man7.org/linux/man-pages/man5/proc.5.html